compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -4104,17 +4105,20 @@ declare_lint! {
     ///          >>> referenced by rust_out.69edbd30df4ae57d-cgu.0
     ///          >>>               rust_out.rust_out.69edbd30df4ae57d-cgu.0.rcgu.o:(rust_out::main::h3a90094b06757803)
     ///   |
-    ///   = note: `#[warn(linker_messages)]` on by default
-    ///
+    /// note: the lint level is defined here
+    ///  --> warn.rs:1:9
+    ///   |
+    /// 1 | #![warn(linker_messages)]
+    ///   |         ^^^^^^^^^^^^^^^
     /// warning: 1 warning emitted
     /// ```
     ///
     /// ### Explanation
     ///
     /// Linkers emit platform-specific and program-specific warnings that cannot be predicted in advance by the rust compiler.
-    /// They are forwarded by default, but can be disabled by adding `#![allow(linker_messages)]` at the crate root.
+    /// They are forwarded by default, but can be disabled by adding `#![allow(linker_messages)]` at the root of a crate that requires linking.
```

Comment:
Shouldn't this be updated to say they are *not* forwarded by default?

Reply:
Good callout. I took the time to also update this paragraph to elaborate on the current state of the lint.